src/penn_chime/models.py

```diff
@@ -94,10 +97,128 @@ def __init__(self, p: Parameters) -> SimSirModel:
         self.dispositions_df = dispositions_df
         self.admits_df = admits_df
         self.census_df = census_df
+
+        if p.n_days_since_first_hospitalized is not None and p.doubling_time is None:
+            # optimize doubling_time
+            argmin_dt = None
+            min_loss = 2.0**99
+            censes = dict()
+            for dt in np.linspace(1,15,29):
+                censes[dt] = self.run_projection(p, dt)
+                self.census_df = censes[dt] # log it into state for loss
+                loss_dt = self.loss_dt(p)
+                if loss_dt < min_loss:
+                    min_loss = loss_dt
+                    argmin_dt = dt
+            self.census_df = censes[dt]
+            p.doubling_time = argmin_dt
+
+            # update all state that is dependent on doubling time.
+            intrinsic_growth_rate = self._intrinsic_growth_rate(p.doubling_time)
+            gamma = 1 / recovery_days
+            beta = self._beta(intrinsic_growth_rate, gamma, susceptible, p.relative_contact_rate)
+            r_t = beta / gamma * susceptible
+            r_naught = (intrinsic_growth_rate + gamma) / gamma
+            doubling_time_t = 1.0 / np.log2(beta * susceptible - gamma + 1)
+            raw_df = sim_sir_df(
+                susceptible,
+                infected,
+                recovered,
+                beta,
+                gamma,
+                p.n_days
```

Comment:
Shouldn't this be p.n_days + p.n_days_since_first_hospitalization?

Reply:
all i'll say is that wouldn't be parity with the notebook. @cjbayesian

Reply:
However, this input can no longer be described as the number of days to project as stated in the ui. It is projecting less than n_days, it is projecting from the first infection. The ui / users here will be confusing / confused.